compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/DataClassMembersGenerator.kt

```diff
@@ -92,7 +92,7 @@ abstract class DataClassMembersGenerator(
             // data classes and corresponding properties can be non-final.
             // We should use getters for such properties (see KT-41284).
             val backingField = property.backingField
-            return if (!forbidDirectFieldAccess && property.modality == Modality.FINAL && backingField != null) {
+            return if (!forbidDirectFieldAccess && irClass.isFinalClass && backingField != null) {
```

Comment:
Could you please comment this change? It influences not only K2 but also K1, so it's not so clear for me.